front/app/modules/commercial/customizable_navbar/admin/containers/HiddenNavbarItemList.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo } from 'react';
+
+// services
+import { addNavbarItem } from '../../services/navbar';
+import { deletePage, IPageData, POLICY_PAGES } from 'services/pages';
+import { getNavbarItemSlug } from 'services/navbar';
+
+// hooks
+import useNavbarItems from 'hooks/useNavbarItems';
+import useRemovedDefaultNavbarItems from '../../hooks/useRemovedDefaultNavbarItems';
+import usePages from 'hooks/usePages';
+import usePageSlugById from 'hooks/usePageSlugById';
+import useLocale from 'hooks/useLocale';
+
+// components
+import { List, Row } from 'components/admin/ResourceList';
+import NavbarItemRow from '../components/NavbarItemRow';
+import { Title } from './VisibleNavbarItemList';
+
+// i18n
+import { injectIntl, FormattedMessage } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+import messages from './messages';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import getItemsNotInNavbar, { IItemNotInNavbar } from './getItemsNotInNavbar';
+
+const POLICY_PAGES_SET = new Set<string>(POLICY_PAGES);
+
+const removePolicyAndProposalsInfoPages = (page: IPageData) => {
+  const code = page.attributes.code;
+
+  return !POLICY_PAGES_SET.has(code) && code !== 'proposals';
+};
+
+const HiddenNavbarItemList = ({
+  intl: { formatMessage },
+}: InjectedIntlProps) => {
+  const navbarItems = useNavbarItems();
+  const removedDefaultNavbarItems = useRemovedDefaultNavbarItems();
+  const pages = usePages();
+  const pageSlugById = usePageSlugById();
+  const locale = useLocale();
+
+  const notAllHooksRendered =
+    isNilOrError(navbarItems) ||
+    isNilOrError(removedDefaultNavbarItems) ||
+    isNilOrError(pages) ||
+    isNilOrError(pageSlugById) ||
+    isNilOrError(locale);
+
+  const itemsNotInNavbar = useMemo(() => {
+    if (notAllHooksRendered) return null;
+
+    return getItemsNotInNavbar(
+      navbarItems,
+      removedDefaultNavbarItems,
+      pages.filter(removePolicyAndProposalsInfoPages)
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [navbarItems, removedDefaultNavbarItems, pages]);
+
+  if (notAllHooksRendered || isNilOrError(itemsNotInNavbar)) {
+    return null;
+  }
+
+  const handleClickAdd = (item: IItemNotInNavbar) => () => {
+    addNavbarItem(item);
+  };
+
+  const handleClickDelete = (pageId?: string) => () => {
+    if (pageId === undefined) return;
+
+    if (window.confirm(formatMessage(messages.deletePageConfirmationHidden))) {
+      deletePage(pageId);
+    }
+  };
+
+  const handleClickView = (item: IItemNotInNavbar) => () => {
+    const originWithLocale = `${window.location.origin}/${locale}`;
+    const slug =
+      getNavbarItemSlug(
+        item.type === 'default_item' ? item.navbarCode : 'custom',
+        pageSlugById,
+        item.type === 'page' ? item.pageId : undefined
+      ) || '/';
+
+    window.open(originWithLocale + slug, '_blank');
```

Comment:
I find `originWithLocale + slug` slightly dangerous. It wouldn't complain if one of the two is null for example. Would make 1 variable of this and then check if it's a string before calling this method.

Reply:
Isn't that what TypeScript is for? To guarantee that both of these are strings?

Reply:
Even better would be to make a function for window.open, so you can do the type checks via the function parameters. If you only do the typeof === checks here, it will just not call the window.open, but we still wouldn't catch the bug unless we stumble upon it.

Reply:
Idk, I think this is going a bit too far. Then you can make the same argument for every single case where we use template strings without checking if the variables in there are `typeof === 'string'`, which is pretty much everywhere

Reply:
Sure, not blocking for me. 

As a PS, we do/did have bugs with URLs not being properly specified (missing / for example), so would not be bad if we did a decent check more often.